Realm.Shared/RealmList.cs

```diff
@@ -32,48 +33,35 @@ namespace Realms
     /// </remarks>
     /// 
     /// <typeparam name="T">Type of the RealmObject which is the target of the relationship.</typeparam>
-    public class RealmList<T> : IList<T> where T : RealmObject
+    [Preserve(AllMembers = true)]
+    public class RealmList<T> : IList<T>, IRealmList, IDynamicMetaObjectProvider where T : RealmObject
     {
-        private class RealmListEnumerator : IEnumerator<T> 
+        public class Enumerator : IEnumerator<T>
         {
-            private int index;
-            private RealmList<T> enumerating;
+            private readonly RealmList<T> _enumerating;
+            private int _index;
 
-            internal RealmListEnumerator(RealmList<T> parent)
+            internal Enumerator(RealmList<T> parent)
             {
-                index = -1;
-                enumerating = parent;
+                _index = -1;
+                _enumerating = parent;
             }
 
             /// <summary>
             /// Return the current related object when iterating a related set.
             /// </summary>
             /// <exception cref="IndexOutOfRangeException">When we are not currently pointing at a valid item, either MoveNext has not been called for the first time or have iterated through all the items.</exception>
-            public T Current
-            {
-                get
-                {
-                    return enumerating[index];
-                }
-            }
+            public T Current => _enumerating[_index];
 
-            // also needed - https://msdn.microsoft.com/en-us/library/s793z9y2.aspx
-            object IEnumerator.Current
-            {
-                get
-                {
-                    return enumerating[index];
-                }
-            }
+            object IEnumerator.Current => Current;
 
             /// <summary>
             ///  Move the iterator to the next related object, starting "before" the first object.
             /// </summary>
             /// <returns>True only if can advance.</returns>
             public bool MoveNext()
             {
-                index++;
-                if (index >= enumerating.Count)
+                if (++_index >= _enumerating.Count)
```

Comment:
I don't fully understand how this is functionally different, but I can revert this change.